src/components/FileDetail/FileDetail.js

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withRouter, Link} from 'react-router-dom';
+import {Mutation} from 'react-apollo';
+import {GET_STUDY_BY_ID} from '../../state/queries';
+import {DELETE_FILE, FILE_DOWNLOAD_URL} from '../../state/mutations';
+import {KF_STUDY_API} from '../../common/globals';
+import TimeAgo from 'react-timeago';
+import {Button, Icon, GridContainer, Avatar} from 'kf-uikit';
+import Badge from '../Badge/Badge';
+import VersionList from '../VersionList/VersionList';
+import UploadVersionContainer from '../../containers/UploadVersionContainer';
+import {
+  fileTypeDetail,
+  formatFileSize,
+  dateCompare,
+} from '../../common/fileMeta';
+/**
+ * Form to add fields to a newly uploaded file
+ */
+
+const FileDetail = ({fileNode, onSubmit, onNameChange, history, match}) => {
+  const sortedVersions =
+    fileNode && fileNode.versions.edges.length > 0
+      ? fileNode.versions.edges.sort(dateCompare)
+      : [];
+  const latestDate =
+    sortedVersions.length > 0 ? sortedVersions[0].node.createdAt : null;
+  const latestSize =
+    sortedVersions.length > 0
+      ? formatFileSize(sortedVersions[0].node.size, true)
+      : 'unknown';
+  const studyId = match.params.kfId;
+  return (
+    <Mutation mutation={FILE_DOWNLOAD_URL}>
+      {downloadFile => (
+        <Mutation
```

Comment:
Let' see if we can move the DELETE_FILE mutation component to it's own component or maybe a HOC. I've noticed the same code in several spots throughtout the file manager and it would nice to have it modularized so we can test it on it's own.